app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkController.kt

```diff
@@ -181,6 +186,37 @@ class DefaultBookmarkController(
         }
     }
 
+    override fun handleSearchStarted() {
+        scope.launch {
+            val allUserNodes = bookmarksStorage
+                ?.getTree(guid = BookmarkRoot.Root.id, recursive = true)
+                ?.let { DesktopFolders(context = activity, showMobileRoot = true).withOptionalDesktopFolders(it) }
+                ?.flattenAllNodes()
+                // Important to strip all default nodes like the desktop toolbar after all children are flattened.
+                ?.filterNot { node -> BookmarkRoot.values().any { it.id == node.guid } }
+
+            store.dispatch(BookmarkFragmentAction.SearchStarted(allUserNodes ?: emptyList()))
+        }
+    }
+
+    override fun handleSearchEnded() {
+        scope.launch {
```

Comment:
How come this needs a scope.launch?

Reply:
Leftover from a previous approach. Will remove. Thanks!